### ISSUE TITLE
for-of says that value can be undefined with strictNullChecks

### ISSUE THREAD

**Issue by RReverser** [Microsoft/TypeScript#8377]
**TypeScript Version:** 

nightly (1.9.0-dev.20160429)

**Code**

```ts
for (let num of [ 1, 2, 3 ]) {
  num.toString();
}
```

**Expected behavior:**
There should be no error; `num` should be reported as just a `number`.

**Actual behavior:** 
`num` has type `number | undefined`, so compiler complains with `Object is possibly 'undefined'`.

**Comment by ivogabe** [comment]
Looks like a duplicate of #8357

**Comment by RReverser** [comment]
Right, thanks!
